lib/datastore/request.js

```diff
@@ -487,6 +487,16 @@ DatastoreRequest.prototype.runQuery = function(query, callback) {
       nextQuery.limit(originalLimitVal);
     }
 
+    if (nextQuery && nextQuery.startVal) {
+      // Convert startVal to base64 format
+      nextQuery.startVal =  new Buffer(nextQuery.startVal).toString('base64');
+    }
+
+    if (nextQuery && nextQuery.endVal) {
+      // Convert endVal to base64 format
+      nextQuery.endVal =  new Buffer(nextQuery.endVal).toString('base64');
+    }
```

Comment:
I think we can simplify this by changing line 468 to:

```js
nextQuery.start(endCursor.toString('base64')).offset(nextOffset);
```

For `endVal` to be set on `nextQuery`, it has to have first been set on the query the user passed in originally. It would be a string already if that's what they had set it as.